### ISSUE TITLE
Issues with Multi-tenant setup for FBIS

### ISSUE THREAD

**Issue by helendallas** [kartoza/django-bims#3940]
This is from the email

Some issues - I am not going create tickets for these.

1. In general number of records differ from FBIS production and new staging. Not sure the reason for this. For example, 795,430 on production, 795,425 on multi-tenant staging.

2. Loading of sites and layers slow -  this seemed to Improve over time.

3. Ecosystem type filter not working and no colour coding of ecosystem types for site ✅

4. No biodiversity modules visible in filter drawer ✅

5. Validation status filter not working - this is also a problem of current production

6. SASS accreditation status - number on new staging differ from production

7. Not all spatial filters re working. For example water source areas - issue already noted on FBIS production.

8. Ecological Category (SASS) filter missing even though it is ticked ion the Site Settings

9. Third party layers - MiniSASS not working (also not working in FBIS production)

10. Third party layers - Inwards wq sites/data not working

11. Not receiving emails for csv or xls downloads - checked spam, nothing ✅

12. Contact For - Recaptha problem - ERROR for site owner: Invalid domain for site key ✅

13. Adding a site - I added a site in SA, but got error message - site is not in the correct country, when I clicked  ✅

14. All Biodiversity buttons missing for Add Fish data, Add anuran data etc. Only SASS, water temp and physics-chem available  ✅

15. Error message : 404-page not found when I added SASS data, then added abiotic data and clicked Submit  ✅

16. Add Source reference for records -  this all need to be sorted alphabetically  (database, report, unpublished data)

17. Error message : 404-page not found when I added source reference data and clicked Submit  ✅

18. When I click on SASS Dashboard of newly created site with SASS data, I get a 404-error  ✅

19. Side panel does not always open and I get a 4040 message, but then cannot get back to site. Have to close and reopen  ✅

20. I should not see “Add SASS data” when adding a wetland site -  I also noted it is still in FBIS production but should be removed. You can only add SASS data to river sites

21. I managed to add physics-chemical data but again on Submit - got error 404 message- page not found; and then return to homepage does not work  ✅

22. Taxon Management - cannot open  ✅

23. Control R results in error 404- message again  ✅

24. I can’t open People/user in BIMS   ✅

25. Validate records - Many unvalidated taxa, See unvalidated data gives me the 404-error message, not working. Filter default should be FBIS, as this is the only data that can be validated

26. Harvest species - should have SA boundary first, not sure what the whole list is, but not needed, only existing spatial boundary we use for FBIS production.

I am going to stop now so that you can check and fix above. Then I will test again.

I have not checked any admin upload taxonomy or occurrence functionality,

**Comment by dimasciput** [comment]
Hi @helendallas,

I'm going to address the issues that exist only in the multi-tenant environment. The issues that occur in production as well will be addressed in their own tickets.

Also, I don't think I will address the different data between staging and production, since the mirroring is not as simple and quick as with the other testing site.

**Comment by dimasciput** [comment]
3. Ecosystem type filter not working and no colour coding of ecosystem types for site 

This is fixed

![image](https://github.com/kartoza/django-bims/assets/1979569/7b0351b7-5ab4-4b52-a355-db196d3d4f04)

![image](https://github.com/kartoza/django-bims/assets/1979569/2f692653-2671-4950-88f4-462047471efd)

**Comment by dimasciput** [comment]
11. Not receiving emails for csv or xls downloads - checked spam, nothing - fixed

![image](https://github.com/kartoza/django-bims/assets/1979569/0a358861-4f1e-46e4-9729-9ccc0f587db1)

**Comment by dimasciput** [comment]
12. Contact For - Recaptha problem - ERROR for site owner: Invalid domain for site key ✅

![image](https://github.com/kartoza/django-bims/assets/1979569/f7e65ee9-8fcd-4a67-beb6-75b716494e43)

@helendallas

**Comment by dimasciput** [comment]
@helendallas most of the 404 issues are fixed.

**Comment by helendallas** [comment]
Thanks @dimasciput Still to attend to before we can migrate

NB: Ecological Category (SASS) filter missing even though it is ticked ion the Site Settings

Other error message may be linked to deployments on staging? Please check.

Error message : 404-page not found when I added source reference data and clicked Submit
Now I get 500 - SERVER ERROR (Only for SASS data when I add source reference), also with physics-chem data. I selected unpublished data

**Comment by helendallas** [comment]
Also @dimasciput Several unvalidated taxa on 
https://fbis.sta.do.kartoza.com/site-visit/list/?validated=[%22unvalidated%22]&o=date&sourceCollection=%5B%22fbis%22%5D

It seems the taxa that should be validated are not. The I can validate site visit data if taxon is not yet validated. Please check and revert

**Comment by dimasciput** [comment]
> Error message : 404-page not found when I added source reference data and clicked Submit
> Now I get 500 - SERVER ERROR (Only for SASS data when I add source reference), also with physics-chem data. I selected unpublished data

I think I fixed the 500 issue for this page.

**Comment by dimasciput** [comment]
> Also @dimasciput Several unvalidated taxa on [https://fbis.sta.do.kartoza.com/site-visit/list/?validated=[%22unvalidated%22]&o=date&sourceCollection=%5B%22fbis%22%5D](https://fbis.sta.do.kartoza.com/site-visit/list/?validated=%5B%22unvalidated%22%5D&o=date&sourceCollection=%5B%22fbis%22%5D)
> 
> It seems the taxa that should be validated are not. The I can validate site visit data if taxon is not yet validated. Please check and revert
> 
> See #3965

@helendallas Yes, currently working on this issue

**Comment by helendallas** [comment]
> > Error message : 404-page not found when I added source reference data and clicked Submit
> > Now I get 500 - SERVER ERROR (Only for SASS data when I add source reference), also with physics-chem data. I selected unpublished data
> 
> I think I fixed the 500 issue for this page.

Great thanks working now

**Comment by dimasciput** [comment]
> @helendallas Fixed for SASS, but not physico-chem. Still get 500 - Server error

This is should be fixed now

**Comment by helendallas** [comment]
When I add physics chemistry data, I still get 500 - server error when I submit

**Comment by dimasciput** [comment]
@helendallas could you test it again please

**Comment by helendallas** [comment]
Working now for physics-chemistry, thanks. Checking water temp again. will make separate ticket if it doesn't work.

**Comment by helendallas** [comment]
I am closing this ticket now as most issues sorted. I created a placeholder ticket for four remaining issues that are also an issue on the current FBIS #4020 

Plus new tickets:
#4019
